src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -381,52 +391,48 @@ DropShards(Oid relationId, char *schemaName, char *relationName,
 		CoordinatedTransactionUse2PC();
 	}
 
-	ShardInterval *shardInterval = NULL;
-	foreach_ptr(shardInterval, deletableShardIntervalList)
+	List *dropTaskList = DropTaskList(relationId, schemaName, relationName,
+									  deletableShardIntervalList);
+
+	Task *task = NULL;
+	foreach_ptr(task, dropTaskList)
 	{
-		uint64 shardId = shardInterval->shardId;
+		uint64 shardId = task->anchorShardId;
 		char *shardRelationName = pstrdup(relationName);
 
-		Assert(shardInterval->relationId == relationId);
-
 		/* build shard relation name */
 		AppendShardIdToName(&shardRelationName, shardId);
 
-		List *shardPlacementList = ShardPlacementList(shardId);
-
 		ShardPlacement *shardPlacement = NULL;
-		foreach_ptr(shardPlacement, shardPlacementList)
+		foreach_ptr(shardPlacement, task->taskPlacementList)
 		{
 			uint64 shardPlacementId = shardPlacement->placementId;
+			int32 shardPlacementGroupId = shardPlacement->groupId;
 
-			if (shardPlacement->groupId == COORDINATOR_GROUP_ID &&
-				IsCoordinator() &&
-				DropSchemaOrDBInProgress())
+			bool isLocalShardPlacement = (shardPlacementGroupId == localGroupId);
```

Comment:
minor: we typically do not prefer `isXXX`, so maybe consider `placementIsLocal ` or such (but it is OK to keep this as well)